pkg/sidecar/util_windows.go

```diff
@@ -6,20 +6,52 @@ package sidecar
 import (
 	"context"
 	"errors"
+	"sync"
 
 	"github.com/spiffe/go-spiffe/v2/workloadapi"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
 // RunDaemon starts the main loop
 // Starts the workload API client to listen for new SVID updates
 // When a new SVID is received on the updateChan, the SVID certificates
 // are stored in disk and a restart signal is sent to the proxy's process
 func (s *Sidecar) RunDaemon(ctx context.Context) error {
```

Comment:
Same comments from the other `RunDaemon` method.

Reply:
I noticed that `RunDaemon` from `util_posix` and `util_windows` are almost exactly the same code, except for the usege of `WithNamedPipeName` or `WithAddr`. Would you look into refactoring the code to reduce this duplication?

Reply:
Yes, they are pretty the same file. Do you suggest do it in this PR? Maybe could be done in other one.

Reply:
I recommend we tackle the refactoring in this PR to keep changes consolidated and prevent duplicating efforts later on.

Reply:
Centralize common logic creating the `RunDaemon` in a new file `util_common.go`:

```
func (s *Sidecar) RunDaemon(ctx context.Context) error {

    addressGetter := s.getAddressGetter() // This will be set in the OS-specific files

    // The rest of the code remains the same...
    // Use addressGetter() wherever you need the address 
}
```

`util_posix.go`:

```
// +build !windows

package sidecar

// OS-specific implementations...

func (s *Sidecar) getAddressGetter() func() string {
    return func() string {
        return "unix://" + s.config.AgentAddress
    }
}

// SignalProcess remains here...

func validateOSConfig(c *Config) error {
    // POSIX-specific validation
}
```

`util_windows.go`

```
// +build windows

package sidecar

// OS-specific implementations...

func (s *Sidecar) getAddressGetter() func() string {
    return func() string {
        return workloadapi.WithNamedPipeName(s.config.AgentAddress)
    }
}

// SignalProcess remains here...

func validateOSConfig(c *Config) error {
    // Windows-specific validation
}
```